packages/shared/src/utils/route/route.tsx

```diff
@@ -0,0 +1,61 @@
+// Checks if pathname matches route. (Works even with query string /?)
+
+// TODO: Add test cases for this
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { deepCopy } from '../object';
+import { routes } from '../routes';
+
+export type TRoute = {
+    component?: React.ElementType | null | ((routes?: TRoute[]) => JSX.Element) | typeof Redirect;
+    default?: boolean;
+    exact?: boolean;
+    getTitle?: () => string;
+    icon_component?: string;
+    id?: string;
+    is_authenticated?: boolean;
+    is_invisible?: boolean;
+    path?: string;
+    to?: string;
+    icon?: string;
+    is_disabled?: boolean;
+    subroutes?: TRoute[];
+    routes?: TRoute[];
+};
+
+type TGetSelectedRoute = {
+    routes: TRoute[];
+    pathname: string;
+};
+
+// @ts-expect-error as this is a utility function with dynamic types
+export const matchRoute = <T,>(route: T, pathname: string) => new RegExp(`^${route?.path}(/.*)?$`).test(pathname);
+
+export const getSelectedRoute = ({ routes, pathname }: TGetSelectedRoute) => {
+    const matching_route = routes.find(route => matchRoute(route, pathname));
+    if (!matching_route) {
+        return routes.find(route => route.default) || routes[0] || null;
+    }
+    return matching_route;
+};
+
+export const isRouteVisible = (route: TRoute, is_logged_in: boolean) =>
+    !(route && route.is_authenticated && !is_logged_in);
+
+export const removePasskeysFromRoutes = (routes_array: TRoute[]) => {
+    return (deepCopy(routes_array) as TRoute[]).filter(route => {
+        if (route.icon === 'IcSecurity') {
```

Comment:
Is there any other identifiers apart from the icon? Using icon names to get the passkey route seems more likely for failure than identifier such as name or path

Reply:
there is no other identifiers in current config.
I think at least just for security object I can add something like id: 'security_routes'